crypto/bls_test.go

```diff
@@ -529,10 +634,24 @@ func TestBatchVerify(t *testing.T) {
 		}
 		valid, err := BatchVerifyBLSSignaturesOneMessage(pks, sigs, input, nil)
 		require.Error(t, err)
-		assert.IsType(t, expectedError, err)
+		assert.True(t, IsInvalidInputsError(err))
 
 		assert.Equal(t, valid, expectedValid,
-			fmt.Sprintf("verification should fail with incorrect input lenghts, got %v", valid))
+			fmt.Sprintf("verification should fail with nil hasher, got %v", valid))
+	})
+
+	// test wrong key
+	t.Run("wrong key", func(t *testing.T) {
+		for i := 0; i < sigsNum; i++ {
+			expectedValid[i] = false
+		}
+		pks[0] = invalidSK(t).PublicKey()
+		valid, err := BatchVerifyBLSSignaturesOneMessage(pks, sigs, input, nil)
+		require.Error(t, err)
+		assert.True(t, IsInvalidInputsError(err))
+
+		assert.Equal(t, valid, expectedValid,
+			fmt.Sprintf("verification should fail with invalid key, got %v", valid))
```

Comment:
```suggestion
		assert.Equalf(t, valid, expectedValid, "verification should fail with invalid key, got %v", valid))
```

Add an `f` to the end of most `assert` or `require` functions, and the final variadic parameter gets treated as a format string and inputs.

Reply:
Nice, I deleted the `Sprintf`s, though I noticed I didn't have to add the `f`.